tools/clang/test/HLSLFileCheck/hlsl/operators/swizzle/swizzleBitfield.hlsl

```diff
@@ -0,0 +1,27 @@
+// RUN: %dxc -T ps_6_6 -HV 2021 %s | FileCheck %s
+
+// This test is a regression test that tests the implementation of L-Value
+// conversions work on bitfields, for both constant and structured buffers
+
+
+struct MyStruct
+{
+    uint v0: 5;
+    uint v1: 15;
+    uint v2: 12;
+};
+
+ConstantBuffer<MyStruct> myConstBuff;
+StructuredBuffer<MyStruct> myStructBuff;
+
+// CHECK: extractvalue %dx.types.ResRet.i32
```

Comment:
Because this isn’t a code generation change I’m actually less concerned that the codegen is correct and more concerned that the AST is correct. Can you write checks to verify that the AST is generating the `LValueToRValue` casts?

I’d look to see 4 AST nodes in the checks for each access:
* The HLSL vector element expression (which will be an rvalue)
* The implicit cast expression for the Vector splat T -> vector<T,1> (which will be an rvalue)
* The implicit cast expression for the LValueToRValue (which will be an rvalue)
* The member expression for the bitfield